src/sentry/static/sentry/app/components/events/interfaces/breadcrumbsV2/breadcrumbs.tsx

```diff
@@ -0,0 +1,332 @@
+      isCollapsed: true,
+    });
+  };
+
+  render() {
+    const {event, type} = this.props;
+    const {breadcrumbFilterGroups, searchTerm} = this.state;
+
+    const {
+      collapsedQuantity,
+      filteredCollapsedBreadcrumbs,
+    } = this.getCollapsedCrumbQuantity();
+
+    return (
+      <EventDataSection
+        type={type}
+        title={
+          <h3>
+            <GuideAnchor target="breadcrumbs" position="bottom">
+              {t('Breadcrumbs')}
+            </GuideAnchor>
+          </h3>
+        }
+        actions={
+          <Search>
+            <BreadcrumbFilter
+              onFilter={this.handleFilter}
+              filterGroups={breadcrumbFilterGroups}
+            />
+            <StyledSearchBar
+              placeholder={t('Search breadcrumbs...')}
+              onChange={this.handleFilterBySearchTerm}
+              query={searchTerm}
+            />
+          </Search>
+        }
+        wrapTitle={false}
+      >
+        <Content>
+          {filteredCollapsedBreadcrumbs.length > 0 ? (
+            <PlatformContextProvider value={{platform: event.platform}}>
+              <BreadCrumbs className="crumbs">
+                {collapsedQuantity > 0 && (
+                  <BreadcrumbCollapsed
+                    onClick={this.handleCollapseToggle}
+                    quantity={collapsedQuantity}
+                  />
+                )}
+                {filteredCollapsedBreadcrumbs.map(
+                  ({color, borderColor, icon, ...crumb}, idx) => {
+                    const Icon = icon as React.ComponentType<IconProps>;
+                    return (
+                      <BreadCrumb
+                        data-test-id="breadcrumb"
+                        key={idx}
+                        hasError={crumb.type === 'message' || crumb.type === 'exception'}
+                      >
+                        <BreadCrumbIconWrapper color={color} borderColor={borderColor}>
+                          <Icon />
+                        </BreadCrumbIconWrapper>
+                        <BreadcrumbRenderer breadcrumb={crumb as Breadcrumb} />
+                        <BreadcrumbTime timestamp={crumb.timestamp} />
+                      </BreadCrumb>
+                    );
+                  }
+                )}
+              </BreadCrumbs>
+            </PlatformContextProvider>
+          ) : (
+            <EmptyStateWarning small>
+              {t('Sorry, no breadcrumbs match your search query.')}
+            </EmptyStateWarning>
+          )}
+        </Content>
+      </EventDataSection>
+    );
+  }
+}
+
+export default BreadcrumbsContainer;
+
+const BreadCrumbs = styled('ul')`
+  padding-left: 0;
+  list-style: none;
+  margin-bottom: 0;
+`;
+
+const Content = styled('div')`
+  border: 1px solid ${p => p.theme.borderLight};
+  border-radius: 3px;
+  box-shadow: 0 1px 2px rgba(0, 0, 0, 0.04);
+  margin-bottom: ${space(3)};
+`;
+
+const Search = styled('div')`
+  display: flex;
+  width: 600px;
```

Comment:
Should we think about mobile experience here?

Reply:
sure!